libflux/src/flux/semantic/builtins.rs

```diff
@@ -301,6 +301,178 @@ pub fn builtins() -> Builtins<'static> {
+                "filter" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        fn: (r: t1) -> bool
+                    ) -> [t2]
+                "#),
+                "first" => Node::Builtin(r#"
+                    forall [t0] where t0: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t0]
+                "#),
+                "float" => Node::Builtin("forall [t0] (v: t0) -> float"),
+                "getColumn" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        table: [t0],
+                        column: string
+                    ) -> [t1]
+                "#),
+                "getRecord" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        table: [t0],
+                        idx: int
+                    ) -> [t1]
+                "#),
+                "group" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?mode: string,
+                        ?columns: [string]
+                    ) -> [t1]
+                "#),
+                "histogram" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string,
+                        ?upperBoundColumn: string,
+                        ?countColumn: string,
+                        bins: [float],
+                        normalize: bool
+                    ) -> [t1]
+                "#),
+                "histogramQuantile" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?quantile: float,
+                        ?countColumn: string,
+                        ?upperBoundColumn: string,
+                        ?valueColumn: string,
+                        ?minValue: float
+                    ) -> [t1]
+                "#),
+                "holtWinters" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?withFit: bool,
+                        ?column: string,
+                        ?timeColumn: string,
+                        n: int,
+                        seasonality: int,
+                        interval: duration
+                    ) -> [t1]
+                "#),
+                "hourSelection" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        start: int,
+                        stop: int,
+                        ?timeColumn: string
+                    ) -> [t1]
+                "#),
```

Comment:
`hourSelection` is essentially a filtering operation. Therefore its input type is just passed through and returned. So it only needs to be parameterized by `t0`.